walk.go

```diff
@@ -9,6 +9,23 @@ import (
 
 const pathSep = "/"
 
+// Escape path component. The special characters ("@" and "\") are escaped to
+// allow mixing the path component with directives (starting with "@") in IPLD
+// data structure.
+func EscapePathComponent(comp string) string {
+	comp = strings.Replace(comp, "\\", "\\\\", -1)
+	comp = strings.Replace(comp, "@", "\\@", -1)
```

Comment:
is it double because json? or what? 

(outside of json, in our own datastruct, we may not need to escape twice?)